Update charactersToDouble() / charactersToFloat() to take in a std::span

cdumez opened the pull request:
#### ea31739e586c9c1c1cde13fbf0e7ff97cfea1b64
<pre>
Update charactersToDouble() / charactersToFloat() to take in a std::span
<a href="https://bugs.webkit.org/show_bug.cgi?id=272012">https://bugs.webkit.org/show_bug.cgi?id=272012</a>

Reviewed by Darin Adler.

* Source/WTF/wtf/JSONValues.cpp:
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::toDouble):
(WTF::StringImpl::toFloat):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::toFloat const):
(WTF::StringView::toDouble const):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::toDoubleType):
(WTF::charactersToDouble):
(WTF::charactersToFloat):
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::parseCSSNumber):
(WebCore::parseSimpleLength):
(WebCore::parseSimpleAngle):
(WebCore::parseSimpleLengthValue):
(WebCore::parseTransformTranslateArguments):
(WebCore::parseTransformAngleArgument):
(WebCore::parseTransformNumberArguments):
* Source/WebCore/css/parser/CSSTokenizerInputStream.cpp:
(WebCore::CSSTokenizerInputStream::getDouble const):
* Source/WebCore/dom/ViewportArguments.cpp:
(WebCore::numericPrefix):
* Source/WebCore/html/track/VTTScanner.cpp:
(WebCore::VTTScanner::scanFloat):
* Source/WebCore/platform/Length.cpp:
(WebCore::parseLength):
(WebCore::newLengthArray):

Canonical link: <a href="https://commits.webkit.org/276933@main">https://commits.webkit.org/276933@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/586949b59596fb6e82bde6db0844490e27ded3a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46191 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25322 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48776 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48863 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42232 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48498 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29654 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22768 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37741 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46769 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22389 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39832 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18949 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19778 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40930 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4235 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/39424 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42486 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41279 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50663 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/45665 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21187 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17701 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44923 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22487 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43838 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22846 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/52816 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6437 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22181 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10812 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/parser/CSSParserFastPaths.cpp:
```diff
@@ -649,7 +648,7 @@ static bool parseTransformTranslateArguments(CharType*& pos, CharType* end, unsi
         CSSUnitType unit = CSSUnitType::CSS_NUMBER;
         double number;
-        if (!parseSimpleLength(pos, argumentLength, unit, number))
+        if (!parseSimpleLength(std::span<const CharType> { pos, argumentLength }, unit, number))
```
Is the template argument just so we can add const? I would have thought we could do this with just braces, or just `std::span` with no template argument.

cdumez replied:
My first attempt was also just curly brackets. However, specifying the span template parameter was indeed required to add the `const`.

cdumez committed:
Update charactersToDouble() / charactersToFloat() to take in a std::span
https://bugs.webkit.org/show_bug.cgi?id=272012

Reviewed by Darin Adler.

* Source/WTF/wtf/JSONValues.cpp:
* Source/WTF/wtf/text/StringImpl.cpp:
(WTF::StringImpl::toDouble):
(WTF::StringImpl::toFloat):
* Source/WTF/wtf/text/StringView.h:
(WTF::StringView::toFloat const):
(WTF::StringView::toDouble const):
* Source/WTF/wtf/text/WTFString.cpp:
(WTF::toDoubleType):
(WTF::charactersToDouble):
(WTF::charactersToFloat):
* Source/WTF/wtf/text/WTFString.h:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::parseCSSNumber):
(WebCore::parseSimpleLength):
(WebCore::parseSimpleAngle):
(WebCore::parseSimpleLengthValue):
(WebCore::parseTransformTranslateArguments):
(WebCore::parseTransformAngleArgument):
(WebCore::parseTransformNumberArguments):
* Source/WebCore/css/parser/CSSTokenizerInputStream.cpp:
(WebCore::CSSTokenizerInputStream::getDouble const):
* Source/WebCore/dom/ViewportArguments.cpp:
(WebCore::numericPrefix):
* Source/WebCore/html/track/VTTScanner.cpp:
(WebCore::VTTScanner::scanFloat):
* Source/WebCore/platform/Length.cpp:
(WebCore::parseLength):
(WebCore::newLengthArray):

Canonical link: https://commits.webkit.org/276933@main
